libsession/src/main/java/org/session/libsession/utilities/GroupUtil.kt

```diff
@@ -30,7 +31,9 @@ object GroupUtil {
 
     @JvmStatic
     fun getEncodedClosedGroupID(groupID: ByteArray): String {
-        return CLOSED_GROUP_PREFIX + Hex.toStringCondensed(groupID)
+        val hex = Hex.toStringCondensed(groupID)
+        if (hex.startsWith(IdPrefix.GROUP.value)) throw IllegalArgumentException("Trying to encode a new closed group")
+        return LEGACY_CLOSED_GROUP_PREFIX + hex
```

Comment:
What is this encoded closed group id? The name is confusing. Does this not need groups v2 data as well?

Reply:
I'm as clueless as you on this one. My **guess** is that they may have stored an "encoded" group id into the db as primary key, and to interact with other part of system you need to decode it back. But it's no longer the case for new groups as we just use the `03xxx` as id directly